python/cudf/dataframe/columnops.py

```diff
@@ -204,6 +205,18 @@ def as_column(arbitrary, nan_as_null=True):
                 mask = cudautils.mask_from_devary(arbitrary)
                 data = data.set_mask(mask)
 
+    elif cuda.is_cuda_array(arbitrary):
+        # Use cuda array interface to do create a numba device array by
+        # reference
+        new_dev_array = cuda.as_cuda_array(arbitrary)
```

Comment:
`arbitrary` is already a cuda array - I don't like that you need `is_cuda_array` followed immediately by `as_cuda_array`. Are you sure that `as_cuda_array` is required here?